### PR TITLE
modify for unsafe

### PR DESCRIPTION
**Pull request by zgfzgf**
#4017 
for var sqi in for statement
unsafe

### CODE REVIEW

File: extern/sector-storage/sched.go [hunk 1]
```diff
@@ -453,11 +456,16 @@ func (sh *scheduler) trySched() {
...
+		rmQueue = append(rmQueue, sqi)
 		scheduled++
 	}
 
+	if len(rmQueue) > 0 {
+		for i := len(rmQueue) - 1; i >= 0; i-- {
+			sh.schedQueue.Remove(rmQueue[i])
+		}
+	}
+
```
Comment (magik6k) [inline extern/sector-storage/sched.go:468]: Was the previous code incorrect?

**Review by magik6k — COMMENTED** [review]
Looks correct, but there are conflicts now (the main issue is now fixed in master it seems)

File: extern/sector-storage/sched.go (hunk 1 above)
Comment (zgfzgf) [reply extern/sector-storage/sched.go:468]: previous code may problem

Comment (magik6k) [reply extern/sector-storage/sched.go:468]: Sure, but what was the issue?

**Comment by zgfzgf** [comment]
hi @magik6k merge ok

File: extern/sector-storage/sched.go (hunk 1 above)
Comment (zgfzgf) [reply extern/sector-storage/sched.go:500]: this makes the code more secure

**Review by magik6k — APPROVED** [review]

**Pull request merged**